python/tk_multi_workfiles/work_area.py

```diff
@@ -190,89 +203,117 @@ def resolve_user_sandboxes(self):
+        extensions = resolved_settings.get("file_extensions") or []
+        extensions = [ext if ext.startswith(".") else ".%s" % ext for ext in extensions if ext]
+        self.valid_file_extensions = extensions
+
+        # test for user sandboxes:
+        self._work_template_contains_user = self.work_template and bool(self._get_template_user_keys(self.work_template))
+        self._publish_template_contains_user = self.publish_template and bool(self._get_template_user_keys(self.publish_template))
+
+    def are_templates_configured(self):
+        """
+        Returns if the templates are all configured for this work area.
+
+        :returns: True if the templates are configured, False otherwise.
+        """
+        if self.work_area_template and self.work_template and self.publish_area_template and self.publish_template:
+            return True
+        else:
+            return False
 
     def _get_settings_for_context(self, context, templates_to_find, settings_to_find=None):
         """
         Find templates for the given context.
+
+        :param context: Toolkit context.
+        :param templates_to_find: Name of the templates to look for in the settings.
+        :param settings_to_find: List of mandatory settings names.
+
+        :returns: A dictionary of the settings. If the dictionary is empty, not settings was found.
+
+        :raises TankError: If no workfiles
         """
         if not context:
             return {}
-        
+
         app = sgtk.platform.current_bundle()
         settings_to_find = settings_to_find or []
-        
+
         resolved_settings = {}
         if app.context == context:
             # no need to look for settings as we already have them in the
             # current environment!
-    
-            # get templates:        
+
+            # get templates:
             for key in templates_to_find:
                 resolved_settings[key] = app.get_template(key)
-                
+
             # get additional settings:
             for key in settings_to_find:
                 resolved_settings[key] = app.get_setting(key)
-                
-        else:        
+
+        else:
             # need to look for settings in a different context/environment
             settings = self._get_raw_app_settings_for_context(app, context)
             if not settings:
-                raise TankError("Failed to find Work Files settings for context '%s' in engine instance '%s'.\n\n"
-                                "Please ensure that the Work Files app is installed for the environment that will"
-                                " be used for this context." % (context, app.engine.instance_name))
-            
+                raise TankError(
+                    "Failed to find the Shotgun File Manager settings for this work area.\n\n"
+                    "Please ensure that the app is installed for the environment that will "
+                    "be used for this work area."
```

Comment:
Is the environment available?  Would be nice to be able to tell somebody that there were no settings for tk-multi-workfiles2 in tk-maya in shot_step.yml.  Especially when they copy/paste the error in a support ticket.